lib/SyTest/Federation/Room.pm

```diff
@@ -167,14 +167,36 @@ sub create_event
       prev_events => make_event_refs( @{ $self->{prev_events} } ),
    );
 
+   $self->_insert_event( $event );
+
    $self->{prev_events} = [ $event ];
 
-   if( defined $fields{state_key} ) {
-      $self->{current_state}{ join "\0", $fields{type}, $fields{state_key} }
+   return $event;
+}
+
+sub insert_event
+{
+   my $self = shift;
+   my ( $event ) = @_;
+
+   $self->_insert_event( $event );
+
+   my $prev_events = $self->{prev_events};
+   push @$prev_events, $event;
+
+   my %remove = map { $_->[0] => 1 } @{ $event->{prev_events} };
```

Comment:
I've stared at these two lines for a while and I can't work out why you're doing them. Could you explain? :)